### PR TITLE
`detect_headers`: sort headers list

### PR DESCRIPTION
**Pull request by t-bltg**
Fix https://github.com/Gnimuc/ClangCompiler.jl/issues/24.

### CODE REVIEW

**Comment by Gnimuc** [comment]
If Julia's `Set` is not deterministic between versions, then it's not a problem of Clang. I'd like to keep the dependency graph minimal.

**Comment by t-bltg** [comment]
No, it's non-deterministic for a single julia version:
```
$ julia generator.jl 1.9
$ julia generator.jl 1.9  # re-run, signatures changed
```

**Comment by Gnimuc** [comment]
Is it same with 1.8?

File: src/generator/resolve_deps.jl [hunk 1]
```diff
@@ -182,7 +182,7 @@ resolve_dependency!(dag::ExprDAG, node::ExprNode{<:AbstractUnionNodeType}, optio
 # at least sort macros according to identifiers, otherwise the order is terribly wrong
 function resolve_dependency!(dag::ExprDAG, node::ExprNode{<:AbstractMacroNodeType}, options)
     # macro name and its arguments are not dependencies
-    args = Set{Symbol}((node.id,))
+    args = OrderedSet{Symbol}((node.id,))
```
Comment (Gnimuc) [inline src/generator/resolve_deps.jl:185]: this container is used as a hash table, the order doesn't matter at all.

**Comment by Gnimuc** [comment]
> How about using a Vector and use unique! afterwards ?
> Or use sort ?

yes, this is a good idea.

File: src/generator/resolve_deps.jl (hunk 1 above)
Comment (t-bltg) [reply src/generator/resolve_deps.jl]: Oh, yeah noticed while reworking.

**Pull request merged**